### PR TITLE
Added flags to configure preamble

### PR DESCRIPTION
**Pull request by alexandervanrenen**
Hi,

I was looking for a logging lib and found this one. It looks great (thanks) and it almost did everything I wanted. But there was no option to configure the preamble (it is a bit to verbose for my purposes), so I added this feature.

Feel free to reject if you don't like it ;)

\- Alex

### CODE REVIEW

File: loguru.hpp [hunk 1]
```diff
@@ -2301,6 +2319,30 @@ namespace loguru
 
 	// ------------------------------------------------------------------------
 
+	static void print_preamble_header(char* out_buff, size_t out_buff_size)
+	{
+		long pos = 0;
+		snprintf(out_buff, out_buff_size, ""); // Make sure there is a '\0' and handle out_buff_size==0
+ 		if(g_preamble_date) {
```
Comment (emilk) [inline loguru.hpp]: Please follow the Loguru code style with a space after if (`if (g_preamble_date) {`) here and elsewhere

**Comment by emilk** [comment]
This looks very nice! I'm gonna take it for a spin to make sure it doesn't degrade performance.

**Comment by emilk** [comment]
This PR causes a minor performance regression with about 10% in buffered mode.

One fix is to use the old code if everything in the preamble is turned on, but I'd rather keep the code readable in this case. If people want the highest possible speed they should be using either `RAW_LOG_F` (which is not slowed down by this PR) – or use another logging library. In other words: I feel the added benefit of this PR out-weights the small performance loss.

**Comment by emilk** [comment]
A smaller concern is how the last `| ` is also controlled by `g_preamble_verbose`.

Example:

`g_preamble_verbose = true`:
```
[ thread name/id ]                   file:line     v| 
[main thread     ]      some_big_file.hpp:12345 WARN| Some warning
```

`g_preamble_verbose = false`:
```
[ thread name/id ]                   file:line     
[main thread     ]      some_big_file.hpp:12345 Some warning
```

If we instead control the last `| ` with a separate flag (`g_preamble_pipe`?) we could have this as:
```
[ thread name/id ]                   file:line  |    
[main thread     ]      some_big_file.hpp:12345 | Some warning
```

This is arguably more readable. Thoughts?

File: loguru.hpp [hunk 2]
```diff
@@ -2301,6 +2319,30 @@ namespace loguru
...
+ 		}
+ 		if(g_preamble_thread) {
+ 			pos += snprintf(out_buff + pos, out_buff_size - pos, "[%-*s]", LOGURU_THREADNAME_WIDTH, " thread name/id");
+ 		}
+ 		if(g_preamble_file) {
+ 			pos += snprintf(out_buff + pos, out_buff_size - pos, "%*s:line     ", LOGURU_FILENAME_WIDTH, "file");
+ 		}
+ 		if(g_preamble_verbose) {
+ 			pos += snprintf(out_buff + pos, out_buff_size - pos, "v| ");
+ 		}
```
Comment (emilk) [inline loguru.hpp]: These new lines have a space before the first tab:

screen shot 2018-01-21 at 21 03 05

**Comment by alexandervanrenen** [comment]
Hi,

Thanks for your feedback.

I fixed the style issues (sorry about these). And I like the idea about controlling the pipe separately. This should all work now. I also fixed a small issue with the spaces of the file column. This should be correct now ;)

About the performance: Whatever you prefer.

- Alex

**Comment by emilk** [comment]
Awesome, thanks!

**Pull request merged**